src/System.Private.CoreLib/src/System/Delegate.cs

```diff
@@ -343,13 +344,13 @@ public static Delegate CreateDelegate(Type type, object target, string method, b
             if (method == null)
                 throw new ArgumentNullException(nameof(method));
 
-            RuntimeType rtType = type as RuntimeType;
+            RuntimeType? rtType = type as RuntimeType;
             if (rtType == null)
                 throw new ArgumentException(SR.Argument_MustBeRuntimeType, nameof(type));
             if (!rtType.IsDelegate())
                 throw new ArgumentException(SR.Arg_MustBeDelegate, nameof(type));
 
-            Delegate d = InternalAlloc(rtType);
+            Delegate? d = InternalAlloc(rtType);
```

Comment:
`InternalAlloc` can never return null.

Reply:
I almost commented on the same thing.  The local here is defined as nullable because it can later be set to null, not because InternalAlloc will return null.

Reply:
Maybe change the set to null to `return null` for clarity?

Reply:
> Maybe change the set to null to return null for clarity?

In general we've been trying to avoid changes that would result in differences in the emited release IL (other than the `null` initialization previously discussed, which would be addressed by the mentioned compiler optimization).  But if you're ok with this change in particular, SGTM.

Reply:
good idea.